eis_toolkit/vector_processing/kriging_interpolation.py

```diff
@@ -0,0 +1,106 @@
+from numbers import Number
+
+import geopandas as gpd
+import numpy as np
+import numpy.ma as ma
+from beartype import beartype
+from beartype.typing import Tuple
+from pykrige.ok import OrdinaryKriging
+from pykrige.uk import UniversalKriging
+
+from eis_toolkit.exceptions import (
+    EmptyDataFrameException,
+    InvalidParameterValueException,
+    NotApplicableGeometryTypeException,
+)
+
+
+def _kriging(
+    data: gpd.GeoDataFrame,
+    resolution: Tuple[Number, Number],
+    extent: Tuple[Number, Number, Number, Number],
+    variogram_model: str,
+    method: str,
+    drift_terms: list,
+) -> Tuple[np.ndarray, dict]:
+
+    coordinates = np.array(list(data.geometry.apply(lambda geom: [geom.x, geom.y, geom.z])))
+    x = coordinates[:, 0]
+    y = coordinates[:, 1]
+    z = coordinates[:, 2]
```

Comment:
I am not sure how often the input data has the interpolation attribute as z-geometry. I think it's more likely that the interpolation attribute is a separate feature, as a column in the geodataframe. I suggest that you either change the function to take a column name for the interpolation or do that and additionally allow for z-geometries to be used for interpolation (this could be a boolean parameter for example).